addon/webextension/selector/uicontrol.js

```diff
@@ -822,6 +821,7 @@ window.uicontrol = (function () {
       primedDocumentHandlers.set(eventName, fn);
     });
     primedDocumentHandlers.set("keyup", keyupHandler);
+    window.addEventListener('beforeunload', beforeunloadHandler);
```

Comment:
the `popstate` event wasn't being triggered by back/forward buttons. `beforeunload` is the best option I found. `unload` seems to be too late.